tests/drivers/counter/rpi_pico_pit/src/main.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2024 Navimatix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zephyr/device.h"
+#include "zephyr/kernel.h"
+#include "zephyr/devicetree.h"
+#include "zephyr/ztest_assert.h"
+#include "zephyr/ztest_test.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/counter.h>
+#include <hardware/pwm.h>
+
+struct counter_pico_pit_fixture {
+	const struct device *pit;
+	const struct device *pit_channel_1;
+	const struct device *pit_channel_2;
+};
+
+const struct device *timer = DEVICE_DT_GET_ONE(raspberrypi_pico_timer);
+
+static void counter_callback_1(const struct device *dev, void *userdata)
+{
+	int *value = userdata;
+
+	*value = 1;
+}
+
+static void counter_callback_2(const struct device *dev, void *userdata)
+{
+	int *value = userdata;
+
+	*value += 1;
+}
+
+static void *counter_pico_pit_setup(void)
+{
+	static struct counter_pico_pit_fixture fixture = {
+		.pit = DEVICE_DT_GET(DT_NODELABEL(rpi_pico_pit_controller)),
+		.pit_channel_1 = DEVICE_DT_GET(DT_NODELABEL(rpi_pico_pit_channel_1)),
+		.pit_channel_2 = DEVICE_DT_GET(DT_NODELABEL(rpi_pico_pit_channel_2)),
+	};
+
+	zassert_not_null(fixture.pit);
+	zassert_not_null(fixture.pit_channel_1);
+	zassert_not_null(fixture.pit_channel_2);
+	return &fixture;
+}
+
+static void counter_pico_pit_before(void *f)
+{
+	struct counter_pico_pit_fixture *fixture = f;
+	struct counter_top_cfg top_cfg = {.callback = NULL, .user_data = NULL, .flags = 0};
+
+	top_cfg.ticks = UINT16_MAX;
+
+	counter_set_top_value(fixture->pit_channel_1, &top_cfg);
+	counter_set_top_value(fixture->pit_channel_2, &top_cfg);
+	counter_stop(fixture->pit_channel_1);
+	counter_stop(fixture->pit_channel_2);
+}
+
+ZTEST_F(counter_pico_pit, test_no_value_change_after_stop)
+{
+	(void)counter_start(fixture->pit_channel_1);
+	k_msleep(100);
+	(void)counter_stop(fixture->pit_channel_1);
```

Comment:
In the test case above the return value of the counter_stop is not being explicitly ignored. I think would be great to follow one pattern.
```suggestion
	zassert_equal(counter_stop(fixture->pit_channel_1),0,"...");
```

Reply:
Use `zassert_ok` instead.

```suggestion
	zassert_ok(counter_stop(fixture->pit_channel_1));
```

Reply:
Yep. counter_pico_pit_before seems to be the only test where it has been missed :)

Reply:
Added the zasserts to before as well.